os-client/src/main/kotlin/io/provenance/os/domain/inputstream/DIMEInputStream.kt

```diff
@@ -177,13 +183,13 @@ class DIMEInputStream(
      * @param keyPair - The encryption key pair to decrypt the stream
      * @return - SignatureInputStream containing a CipherInputStream in DECRYPT mode.
      */
-    fun getDecryptedPayload(keyPair: KeyPair): SignatureInputStream {
+    fun getDecryptedPayload(encryptionKeyRef: KeyRef, signer: SignerImpl): SignatureInputStream {
         // seek past the header
         pos += header.size
 
-        return ProvenanceDIME.getDEK(dime.audienceList, keyPair)
+        return ProvenanceDIME.getDEK(dime.audienceList, encryptionKeyRef)
             .let { ProvenanceDIME.decryptPayload(this, it) }
-            .verify(getFirstSignaturePublicKey(), getFirstSignature())
+            .verify(signer, getFirstSignaturePublicKey(), getFirstSignature())
```

Comment:
where is this verify function? I just wanted to see what it is doing now but can't find it.

Reply:
https://github.com/provenance-io/p8e/blob/5eb9b7eb60fbb70537faa9466602104f290c77d4/p8e-util/src/main/kotlin/io/p8e/crypto/SignatureInputStream.kt#L56

Reply:
thanks!